### PR TITLE
Remove some xfails in sorting tests

### PR DESCRIPTION
**Pull request by asi1024**
Split from #3155.

### CODE REVIEW

File: cupy/core/_routines_sorting.pyx [hunk 1]
```diff
@@ -69,8 +69,7 @@ cdef ndarray _ndarray_argsort(ndarray self, axis):
                            'reinstall CuPy after uninstalling it.')
 
     if ndim == 0:
-        raise ValueError('Sorting arrays with the rank of zero is not '
-                         'supported')  # as numpy.argsort() raises
+        return cupy.zeros((1,), dtype=cupy.int64)
```
Comment (toslunar) [inline cupy/core/_routines_sorting.pyx]: The fix is beyond `cat:test`.

- `cupy.ones(()).argsort(axis=2)` should raise `AxisError`.  How about using `atleast_1d`?
- `argmin(axis=0)` and `argmax(axis=0)` have the same issue.

Comment (asi1024) [reply cupy/core/_routines_sorting.pyx]: Fixed!

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Chainer flaky test?
```

=================================== FAILURES ===================================
_ TestMultiprocessIteratorStalledDatasetDetection_param_1_{nth=1}.test_stalled_getitem _

self = <chainer.testing._bundle.TestMultiprocessIteratorStalledDatasetDetection_param_1_{nth=1} testMethod=test_stalled_getitem>

    def test_stalled_getitem(self):
        nth = self.nth
        batch_size = 2
        sleep = 0.5
        timeout = 0.1
    
        dataset = StallingDataset(nth, sleep)
        it = iterators.MultiprocessIterator(
            dataset, batch_size=batch_size, shuffle=False,
            dataset_timeout=timeout, repeat=False)
    
        # TimeoutWarning should be issued.
        warning_cls = iterators.MultiprocessIterator.TimeoutWarning
        data = []
        # No warning until the stalling batch
        for i in range(nth // batch_size):
            data.append(it.next())
        # Warning on the stalling batch
        with testing.assert_warns(warning_cls):
            data.append(it.next())
        # Retrieve data until the end
        while True:
            try:
>               data.append(it.next())

batch_size = 2
data       = [[0, 1]]
dataset    = <chainer_tests.iterators_tests.test_multiprocess_iterator.StallingDataset object at 0x7fc733df46a0>
it         = <chainer.iterators.multiprocess_iterator.MultiprocessIterator object at 0x7fc733df4e10>
nth        = 1
self       = <chainer.testing._bundle.TestMultiprocessIteratorStalledDatasetDetection_param_1_{nth=1} testMethod=test_stalled_getitem>
sleep      = 0.5
timeout    = 0.1
warning_cls = <class 'chainer.iterators.multiprocess_iterator.MultiprocessIterator.TimeoutWarning'>

tests/chainer_tests/iterators_tests/test_multiprocess_iterator.py:847: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
chainer/iterators/multiprocess_iterator.py:155: in __next__
    batch, state = self._comm.get()
chainer/iterators/multiprocess_iterator.py:333: in get
    _raise_timeout_warning()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def _raise_timeout_warning():
        warnings.warn(
            'Stalled dataset is detected. '
            'See the documentation of MultiprocessIterator for common causes and '
            'workarounds:\n'
            'https://docs.chainer.org/en/stable/reference/generated/'
            'chainer.iterators.MultiprocessIterator.html',
>           MultiprocessIterator.TimeoutWarning)
E       chainer.iterators.multiprocess_iterator.MultiprocessIterator.TimeoutWarning: Parameterized test failed.
E       
E       Base test method: TestMultiprocessIteratorStalledDatasetDetection.test_stalled_getitem
E       Test parameters:
E         nth: 1
E       
E       
E       (caused by)
E       TimeoutWarning: Stalled dataset is detected. See the documentation of MultiprocessIterator for common causes and workarounds:
E       https://docs.chainer.org/en/stable/reference/generated/chainer.iterators.MultiprocessIterator.html
```

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by takagi** [comment]
Let me backport this PR for #3291 .

**Pull request merged**